addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -2416,6 +2416,122 @@ QUnit.test('chat window should remain folded when new message is received', asyn
     );
 });
 
+QUnit.test('chat window scroll position should remain the same after switching left', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        is_minimized: true,
+        state: 'open',
+    });
+    this.data['mail.channel'].records.push({
+        id: 21,
+        is_minimized: true,
+        state: 'open',
+    });
+    for (let i = 0; i < 10; i++) {
+        this.data['mail.message'].records.push({
+            body: "not empty",
+            channel_ids: [20],
+        });
+    }
+    for (let i = 0; i < 10; i++) {
+        this.data['mail.message'].records.push({
+            body: "not empty",
+            channel_ids: [21],
+        });
+    }
+    await this.start();
+
+    const thread1LocalId = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel',
+    }).localId;
+    const thread2LocalId = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 21,
+        model: 'mail.channel',
+    }).localId;
```

Comment:
Note during forward-port: use `pyEnv` and replace reliance of `this` by getting things returned by `start` (e.g. `messaging`).